commit-status-publisher-server/src/main/java/jetbrains/buildServer/commitPublisher/github/ChangeStatusUpdater.java

```diff
@@ -243,15 +243,22 @@ private String getComment(@NotNull RepositoryVersion version,
               }
             }
 
-            if (artifacts != null && !artifacts.isEmpty()) {
-              comment.append("\n");
-              if (build.getBuildStatus() != Status.NORMAL) {
-                comment.append("Build status is not normal, not calculating artifact size\n");
-              } else {
-                comment.append(buildSize.getComment(build));
+            try {
+              String baseBranch = api.findBasePullRequestLabel(repositoryOwner,
+                                                               repositoryName, version.getVcsBranch());
+              if (baseBranch != null && baseBranch.equals(Constants.GITHUB_MASTER_BRANCH)) {
+                if (artifacts != null && !artifacts.isEmpty()) {
+                  comment.append("\n");
+                  if (build.getBuildStatus() != Status.NORMAL) {
+                    comment.append("Build status is not normal, not calculating artifact size\n");
+                  } else {
+                    comment.append(buildSize.getComment(build));
+                  }
+                }

```

Comment:
If the branches don't match we should do another comment.append("Not merging into master, not calculating artifact size\n") or something like that